tests/test_rad.py

```diff
@@ -20,6 +63,82 @@ def test_landsat_metadata(self):
             datetime(2020, 1, 28, 0, 0, tzinfo=timezone.utc),
         )
 
+    def test_angles(self):
+        subsample = 10
+        nodata = -32768
+        resampling = 'bilinear'
+        df = get_pathrow()
+        with tempfile.TemporaryDirectory() as gz_tmp:
+            angle_paths = extract_espa_tools(gz_tmp)
+            with tempfile.TemporaryDirectory() as tmp_full:
+                (
+                    out_path,
+                    angle_paths_out,
+                    l57_angles_path,
+                    l8_angles_path,
+                ) = landsat_angle_prep(
+                    ref_file=l7_225078_20110306_B1,
+                    out_dir=tmp_full,
+                    l57_angles_path=str(angle_paths.l57_angles_path),
+                    l8_angles_path=str(angle_paths.l8_angles_path),
+                )
+                angle_paths_in = run_espa_command(
+                    angle_paths_out.vaa,
+                    str(l7_225078_20110306_ang),
+                    'l7',
+                    str(angle_paths.l57_angles_path),
+                    str(angle_paths.l8_angles_path),
+                    subsample,
+                    out_path,
+                    0,
+                )
+                angle_array_resamp_da = open_angle_file(
+                    angle_paths_in.solar,
+                    512,
+                    angle_paths_in.out_order['zenith'],
+                    nodata,
+                    subsample,
+                )
+                # Used to generate test data
+                # save_angle_data(angle_array_resamp_da, angle_paths_in.solar, nodata)
+                data = (
+                    angle_array_resamp_da.fillna(nodata)
+                    .gw.assign_nodata_attrs(nodata)
+                    .astype('int32')
+                )
+                # Check against a full solar angle image
+                with gw.open(l7_225078_20110306_SZA, chunks=data.gw.row_chunks) as src:
+                    self.assertTrue(src.equals(data))

```

Comment:
This should catch future issues, as it compares against an angle raster that correctly covers a full WRS2 path/row.